Sources/SwiftLexicalLookup/ScopeImplementations.swift

```diff
@@ -265,15 +312,57 @@ import SwiftSyntax
   /// }
   /// // a is visible here
   /// ```
-  @_spi(Experimental) public func lookup(
-    for name: String?,
-    at syntax: SyntaxProtocol,
-    with config: LookupConfig
+  @_spi(Experimental) public func introducesToSequentialParent(
+    for identifier: Identifier?,
+    at origin: SyntaxProtocol,
+    with config: LookupConfig,
+    state: LookupState
   ) -> [LookupResult] {
-    if body.position <= syntax.position && body.endPosition >= syntax.position {
-      return lookupInParent(for: name, at: self, with: config)
+    guard body.position > origin.position || body.endPosition < origin.position
+    else { return [] }
+
+    let names = conditions.flatMap { element in
+      LookupName.getNames(from: element.condition, accessibleAfter: element.endPosition)
+    }.filter { introducedName in
+      checkName(identifier, refersTo: introducedName, at: origin)
+    }
+
+    return names.isEmpty ? [] : [.fromScope(self, withNames: names)]
+  }
+
+  @_spi(Experimental) public var introducedNames: [LookupName] {
+    []
+  }
+}
+
+@_spi(Experimental) extension ActorDeclSyntax: TypeScopeSyntax {}
+@_spi(Experimental) extension ClassDeclSyntax: TypeScopeSyntax {}
+@_spi(Experimental) extension StructDeclSyntax: TypeScopeSyntax {}
+@_spi(Experimental) extension EnumDeclSyntax: TypeScopeSyntax {}
+@_spi(Experimental) extension ExtensionDeclSyntax: TypeScopeSyntax {}
+
+@_spi(Experimental) extension AccessorDeclSyntax: ScopeSyntax {
+  /// Implicit and/or explicit names introduced
+  /// within the accessor.
+  @_spi(Experimental) public var introducedNames: [LookupName] {
+    if let parameters {
+      return LookupName.getNames(from: parameters)
     } else {
-      return defaultLookupImplementation(for: name, at: syntax, with: config)
+      switch accessorSpecifier.tokenKind {
+      case .keyword(.set), .keyword(.willSet):
+        return [.implicit(.newValue(self))]
+      case .keyword(.didSet):
+        return [.implicit(.oldValue(self))]
```

Comment:
All these accessors can also specify custom names instead of using the implicit `newValue` or `oldValue`. Eg. 

```swift
var x: Int = 2 {
  didSet(myNewValue) {
    print(myNewValue)
  }
}
```

Reply:
It already actually [works](https://github.com/swiftlang/swift-syntax/blob/cf273f8943c52a8df8c7348c81a5120b16c577c7/Tests/SwiftLexicalLookupTest/NameLookupTests.swift#L649-L651), as we first try to [extract names from `parameters` if available ](https://github.com/swiftlang/swift-syntax/blob/cf273f8943c52a8df8c7348c81a5120b16c577c7/Sources/SwiftLexicalLookup/ScopeImplementations.swift#L347).